### PR TITLE
[IMP] helpdesk_fieldservice (Close Ticket Wizard)

### PR DESCRIPTION
**Pull request by patrickrwilson**
Adds a close ticket wizard that is triggered when the fso is being completed and the related ticket is still open.

### CODE REVIEW

File: helpdesk_fieldservice/wizard/fsm_order_close_wizard.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    resolution = fields.Text(string="Resolution")
+    team_id = fields.Many2one('helpdesk.team', string="Helpdesk Team")
+    stage_id = fields.Many2one('helpdesk.stage', string="Stage")
+    ticket_id = fields.Many2one('helpdesk.ticket', string="Ticket")
+
+    @api.multi
+    def action_close_ticket(self):
+        for record in self:
+            if record.ticket_id.stage_id.is_close is False:
```
Comment (max3903) [inline helpdesk_fieldservice/wizard/fsm_order_close_wizard.py]:
```suggestion
            if not record.ticket_id.stage_id.is_close:
```

Comment (max3903) [reply helpdesk_fieldservice/wizard/fsm_order_close_wizard.py]: @patrickrwilson What if the ticket has multiple FSM orders and one of them is not closed?

File: helpdesk_fieldservice/wizard/fsm_order_close_wizard.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                record.ticket_id.write({'resolution': record.resolution,
+                                        'stage_id': record.stage_id.id})
+        return {
+            'type': 'ir.actions.client',
+            'tag': 'reload',
+        }
+
+class FSMOrder(models.Model):
```
Comment (max3903) [inline helpdesk_fieldservice/wizard/fsm_order_close_wizard.py]: @patrickrwilson This class can go in `models/fsm_order.py`.

File: helpdesk_fieldservice/wizard/fsm_order_close_wizard.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                         domain="[('team_ids', '=', team_id)]"
+                         options="{'no_create': True, 'no_open': True, 'no_create_edit': True}"/>
+              </group>
+              <group>
+                  <field name="resolution"
+                         string="Ticket Resolution"
+                         required="1"/>
+              </group>
+              <footer>
+                  <button string="Update Ticket"
```
Comment (max3903) [inline helpdesk_fieldservice/wizard/fsm_order_close_wizard.xml]:
```suggestion
                  <button string="Close Ticket"
```

File: helpdesk_fieldservice/wizard/fsm_order_close_wizard.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2019 Pavlov Media
+# License Proprietary. Do not copy, share nor distribute.
```
Comment (max3903) [inline helpdesk_fieldservice/wizard/fsm_order_close_wizard.py]:
```suggestion
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```

File: helpdesk_fieldservice/wizard/fsm_order_close_wizard.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<!-- Copyright (C) 2019 Pavlov Media
+    License Proprietary. Do not copy, share nor distribute.-->
```
Comment (max3903) [inline helpdesk_fieldservice/wizard/fsm_order_close_wizard.xml]:
```suggestion
    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
```

File: helpdesk_fieldservice/wizard/__init__.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
```
Comment (max3903) [inline helpdesk_fieldservice/wizard/__init__.py]:
```suggestion
# Copyright (C) 2019 Pavlov Media
```

File: helpdesk_fieldservice/wizard/fsm_order_close_wizard.py (hunk 1 above)
Comment (patrickrwilson) [reply helpdesk_fieldservice/wizard/fsm_order_close_wizard.py]: good point, i'll include that logic

**Review by osi-scampbell — CHANGES_REQUESTED** [review]
In helpdesk_fieldservice/models/fsm_order.py we need to import ValidationError and _ or else it will throw an error, also remove those objects from the import list on fsm_order_close_wizard.py because we don't use them on that file

**Review by max3903 — APPROVED** [review]

**Comment by max3903** [comment]
@patrickrwilson Can you fix the Travis errors?
https://travis-ci.org/ursais/osi-addons/jobs/611966871#L327

Thanks!

**Comment by osi-scampbell** [comment]
> @patrickrwilson Can you fix the Travis errors?
> https://travis-ci.org/ursais/osi-addons/jobs/611966871#L327
> 
> Thanks!

@patrickrwilson If you import those two objects I mentioned in my review it will take care of everything except helpdesk_fieldservice/models/fsm_order.py line 32. Looks like that line is 1 character too long. Then the travis should be green!

**Review by osi-scampbell — APPROVED** [review]

**Comment by patrickrwilson** [comment]
@max3903 @osi-scampbell Thanks for the reviews, i just sent the last commit which i think has all the fixes that you brought up.

**Pull request merged**